File: data-exercise/spark/spark-sql/src/main/java/com/mqjd/spark/spark01/HelloSparkDF.java
package com.mqjd.spark.spark01;

import com.mqjd.spark.sql.base.ParameterTool;
import com.mqjd.spark.sql.base.SparkDFBase;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SaveMode;
import org.apache.spark.sql.SparkSession;

public class HelloSparkDF extends SparkDFBase {

    public static void main(String[] args) {
        ParameterTool parameterTool = ParameterTool.fromArgs(args);
        String output = parameterTool.get(ParameterTool.OUTPUT());
        SparkSession spark = createSession("HelloSparkDF");
        Dataset<Row> df = spark.range(1, 10).toDF("number");
        Dataset<Row> mapped_df = df.select(df.col("number").plus(10));
        mapped_df.write().option("delimiter", ",").option("header", "true").mode(SaveMode.Overwrite)
            .csv(output);
        spark.stop();
    }

}

File: data-exercise/spark/spark-sql/src/main/scala/com/mqjd/spark/scala/spark01/HelloSparkDF.scala
package com.mqjd.spark.scala.spark01

import com.mqjd.spark.scala.base.ParameterTool
import com.mqjd.spark.scala.base.SparkDFBase
import org.apache.spark.sql.SaveMode
import org.apache.spark.sql.SparkSession

object HelloSparkDF extends SparkDFBase {
  def main(args: Array[String]): Unit = {
    val spark: SparkSession = createSparkSession()
    val parameterTool = ParameterTool.fromArgs(args)
    val output = parameterTool.get(ParameterTool.OUTPUT)
    val df = spark.range(1, 10).toDF("number")
    val mapped_df = df.select((df.col("number") + 10).alias("number"))
    mapped_df.write.option("delimiter", ",").option("header", "true").mode(SaveMode.Overwrite).csv(output)
    spark.stop()
  }
}
